linked-list/src/example/java/Element.java

```diff
@@ -0,0 +1,52 @@
+public final class Element {
+    private final Object value;
+    private Element prev;
+    private Element next;
+
+    public Element(Object value, Element prev, Element next) {
+        this.value = value;
+        this.prev = prev;
+        this.next = next;
+    }
+
+    public Element getPrev() {
+        return prev;
+    }
+
+    public void setPrev(Element prev) {
+        this.prev = prev;
+    }
+
+    public Element getNext() {
+        return next;
+    }
+
+    public void setNext(Element next) {
+        this.next = next;
+    }
+
+    public Object getValue() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
You also argue that we unwrap the next/prev fields into public-scoped fields and use them directly since `Element` has tighter scoping. I'm don't have a strong opinion either way.